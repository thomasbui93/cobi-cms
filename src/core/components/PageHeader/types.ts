export interface InterfacePageHeaderProps {
    title: string,
    description: string,
    meta: InterfacePageMeta,
    classes?: string | string[]
}

export interface InterfacePageMeta {
    title: string,
    description: string
}