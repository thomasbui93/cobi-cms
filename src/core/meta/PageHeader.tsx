import * as React from 'react'
import { Helmet } from 'react-helmet'

export interface InterfacePageHeaderProps {
    title: string,
    description: string,
    meta: InterfacePageMeta,
    classes?: string | string[]
}

export interface InterfacePageMeta {
    title: string,
    description: string
}

export class PageHeader extends React.Component<InterfacePageHeaderProps> {
    constructor(props: InterfacePageHeaderProps) {
        super(props)
        this.getClassName = this.getClassName.bind(this)
    }
    public getClassName():string {
        return this.props.classes ? (
            typeof this.props.classes === 'string' ? this.props.classes : this.props.classes.join(' ')
        ) : ''
    }
    public render() {
        return (
            <div className={`page-header ${this.getClassName()}`}>
                <h1 className='page-header__title'>
                    {this.props.title}
                </h1>
                <p className='page-header__description'>
                    {this.props.description}
                </p>
                <Helmet>
                    <title>{this.props.meta.title}</title>
                    <meta name='description' content={this.props.meta.description} />
                </Helmet>
            </div>
        )
    }
}