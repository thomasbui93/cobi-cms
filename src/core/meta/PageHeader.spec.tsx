import * as React from 'react'
import * as renderer from 'react-test-renderer'
import { PageHeader } from './PageHeader'
import { shallow } from 'enzyme'
import { Helmet } from 'react-helmet'

describe('PageHeader component', () => {
    const examplePageHeader = {
        title: 'Home',
        description: 'Home Page',
        meta: {
            title: 'Home',
            description: 'Home Page'
        },
        classes: 'sample'
    }

    it('renders correctly', () => {
        const tree = renderer
            .create(<PageHeader
                title={examplePageHeader.title}
                description={examplePageHeader.description}
                meta={examplePageHeader.meta}
                classes={examplePageHeader.classes}
            />)
            .toJSON()
        expect(tree).toMatchSnapshot()
    })

    it('should contain Helmet, title and description', () => {
        const header = shallow(<PageHeader
            title={examplePageHeader.title}
            description={examplePageHeader.description}
            meta={examplePageHeader.meta}
            classes={examplePageHeader.classes}
        />)
        expect(header.find(Helmet).length).toBe(1)
        expect(header.find('.page-header__title').length).toBe(1)
        expect(header.find('.page-header__description').length).toBe(1)
    })

    it('getClassName should return associated string if classes props is string or none', () => {
        const header = shallow(<PageHeader
            title={examplePageHeader.title}
            description={examplePageHeader.description}
            meta={examplePageHeader.meta}
            classes={examplePageHeader.classes}
        />)
        expect((header.instance() as PageHeader).getClassName()).toBe('sample')
    })

    it('getClassName should return concat class names if classes props is an array', () => {
        const header = shallow(<PageHeader
            title={examplePageHeader.title}
            description={examplePageHeader.description}
            meta={examplePageHeader.meta}
            classes={['sa', 'mp', 'le']}
        />)
        expect((header.instance() as PageHeader).getClassName()).toBe('sa mp le')
    })
})