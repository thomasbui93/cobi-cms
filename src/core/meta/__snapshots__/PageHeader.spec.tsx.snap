// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageHeader component renders correctly 1`] = `
<div
  className="page-header sample"
>
  <h1
    className="page-header__title"
  >
    Home
  </h1>
  <p
    className="page-header__description"
  >
    Home Page
  </p>
</div>
`;
